editor/src/components/canvas/canvas-strategies/strategies/reparent-metastrategy.tsx

```diff
@@ -106,6 +123,80 @@ export function getApplicableReparentFactories(
   return factories
 }
 
+function getStartingTargetParentsToFilterOutInner(
+  canvasState: InteractionCanvasState,
+  interactionSession: InteractionSession,
+): ReparentTargetsToFilter | null {
+  if (isInsertionSubjects(canvasState.interactionTarget)) {
+    return null
+  }
+
+  const interactionData = interactionSession.interactionData
+  if (interactionData.type !== 'DRAG') {
+    throw new Error(
+      `getStartingTargetParentsToFilterOut should only be called from a DRAG type interaction, not a ${interactionData.type} type interaction`,
+    )
+  }
+
+  const pointOnCanvas = interactionData.originalDragStart
+  const allowSmallerParent = interactionData.modifiers.cmd
+    ? 'allow-smaller-parent'
+    : 'disallow-smaller-parent'
+
+  const reparentSubjects = reparentSubjectsForInteractionTarget(canvasState.interactionTarget)
+
+  const strictBoundsResult = getReparentTargetUnified(
+    reparentSubjects,
+    pointOnCanvas,
+    interactionData.modifiers.cmd,
+    canvasState,
+    canvasState.startingMetadata,
+    canvasState.startingAllElementProps,
+    'use-strict-bounds',
+    allowSmallerParent,
+  )
+
+  const missingBoundsResult = getReparentTargetUnified(
+    reparentSubjects,
+    pointOnCanvas,
+    interactionData.modifiers.cmd,
+    canvasState,
+    canvasState.startingMetadata,
+    canvasState.startingAllElementProps,
+    'allow-missing-bounds',
+    allowSmallerParent,
+  )
+
+  return reparentTargetsToFilter(strictBoundsResult, missingBoundsResult)
+}
+
+function isCanvasState(
+  v: InteractionCanvasState | InteractionSession,
```

Comment:
```suggestion
  canvasState: InteractionCanvasState | InteractionSession,
```

Reply:
I'm going to go with `value` because we don't know if it is or isn't a canvas state yet!